src/driver.rs

```diff
@@ -263,16 +264,21 @@ impl QldbDriver {
                             // might fail ("transaction already open"). This is not a safety issue, but it will cause user-level failures as BadRequests are typically not retried.
                             //
                             // So, we take the pragmatic approach and mark the session as invalid. This prevents it being returned to the pool.
-                            if let Err(e) = self.client.abort_transaction(session_token).await {
-                                debug!("ignoring failure to abort tx {}: {}", a.tx_id, e);
-                                session_handle.notify_invalid();
-                            }
+                            match self.client.abort_transaction(session_token).await {
+                                Ok(r) => {
+                                    execution_stats.accumulate(&r);
+                                }
+                                Err(e) => {
+                                    debug!("ignoring failure to abort tx {}: {}", a.tx_id, e);
+                                    session_handle.notify_invalid();
+                                }
+                            };
 
                             // If a user calls abort, they will get `Ok(user_data)` returned. This does not allow a disambiguation between data that was committed (verified by OCC) versus
                             // data that was captured in an aborted transaction. Consider an alternative API where the Result Ok variant is an Enum with either `Committed(data)` or
                             // `Uncommitted(data)`. More clear, but also much more annoying to work with. Because this API is generic over `R`, we leave the commit/abort wrapping up to
                             // users.
-                            return Ok(a.user_data);
+                            return Ok(a.user_data); // TODO: return stats
```

Comment:
Any reason not to do this now?

Reply:
Working on it. 

The issue is not 'add return foo' and remove the TODO. It's a bit deeper. I'm not even sure what to return (e.g. is it a sum of all the timing information and IO usage for all requests across all retries), but I do know the code is not appropriately factored to make that possible.